src/consensus.rs

```diff
@@ -41,7 +42,7 @@ pub struct Consensus<TYPES: NodeType> {
     /// In the future we will need a different struct similar to VidDisperse except
     /// it stores only one share.
     /// TODO <https://github.com/EspressoSystems/HotShot/issues/1732>
-    pub vid_shares: BTreeMap<TYPES::Time, Proposal<TYPES, VidDisperse<TYPES>>>,
+    pub vid_shares: BTreeMap<TYPES::Time, Proposal<TYPES, VidDisperseShare<TYPES>>>,
```

Comment:
I think we want to keep this as the full disperse.  The libp2p request response protocol dictates that the leader and DA committee should store all of the shares so they can send one share to anyone who asks.

Reply:
hmm we probably need to resolve this issue some other way.  For now this is totally fine.   I'm pretty sure the signature is over the `vid_commitment` only so it should be valid for any Share or Disperse but we can refine after your solution goes unblocks the hotshot changes.

`pub vid_shares: BTreeMap<TYPES::Time, HashMap<TYPES::SignatureKey, Proposal<TYPES, VidDisperseShare<TYPES>>>>`, so let's just go with this

Reply:
Let me know if you need help debugging the tests